smarts/core/default_map_builder.py

```diff
@@ -23,11 +23,61 @@
 from typing import NamedTuple, Tuple
 from pathlib import Path
 from smarts.core.road_map import RoadMap
-from smarts.core.utils.file import file_md5_hash
+from smarts.core.utils.file import file_md5_hash, path2hash
+
 
 _existing_map = None
 
 
+def _cache_result(map_spec, road_map, road_map_hash: str):
+    global _existing_map
+    from smarts.sstudio.types import MapSpec
+
+    class _RoadMapInfo(NamedTuple):
+        map_spec: MapSpec
+        obj: RoadMap
+        map_hash: str
+
+    _existing_map = _RoadMapInfo(map_spec, road_map, road_map_hash)
+
+
+def _clear_cache():
+    global _existing_map
+    if _existing_map:
+        import gc
+
+        # Try to only keep one map around at a time...
+        del _existing_map
+        _existing_map = None
+        gc.collect
+
+
+_UNKNOWN_MAP = 0
+_SUMO_MAP = 1
+_OPENDRIVE_MAP = 2
+
+
+def _find_mapfile_in_dir(map_dir: str) -> Tuple[int, str]:
+    map_filename_type = {
+        "map.net.xml": _SUMO_MAP,
+        "shifted_map-AUTOGEN.net.xml": _SUMO_MAP,
+        "map.xodr": _OPENDRIVE_MAP,
+    }
+    map_type = _UNKNOWN_MAP
+    map_path = map_dir
+    for f in os.listdir(map_dir):
+        cand_map_type = map_filename_type.get(f)
+        if cand_map_type is not None:
+            return cand_map_type, os.path.join(map_dir, f)
+        if f.endswith(".net.xml"):
+            map_type = _SUMO_MAP
+            map_path = os.path.join(map_dir, f)
+        elif f.endswith(".xodr"):
+            map_type = _OPENDRIVE_MAP
+            map_path = os.path.join(map_dir, f)
```

Comment:
Very minor, but maybe we want to return here since this could continue looping needlessly if we've already found a map file.

Reply:
Unfortunately we can't do that b/c we need to make sure we don't have a file like `map.net.xml` later in the directory list.   (If we do, we want to return that one instead.  Actually that was the main point of the original change.)